functional_programming_tests/test/main.cpp

```diff
@@ -32,7 +32,22 @@ class Map {
   T at(Position const& pos) const { return data_.at(pos.y).at(pos.x); }
   std::vector<std::vector<T>>& get_data() { return data_; }
   const std::vector<std::vector<T>>& get_data() const { return data_; }
-  bool empty() { return data_.empty(); }
+
+  /**
+   * @brief      Checks if the map is empty
+   * @return     bool indicating if the map is empty
+   */
+  bool empty() const { return data_.empty(); }
+
+  /**
+   * @brief      Gets the dimensions of the map
+   * @return     std::pair containing the shape of the map
+   *            first: number of rows
+   *            second: number of columns
+   */
+  std::pair<size_t, size_t> shape() const {
+    return std::make_pair(data_.size(), data_.at(0).size());
+  }
```

Comment:
Good idea

Reply:
Wasn't my idea. You had it in some other code.